dask/utils.py

```diff
@@ -399,14 +402,21 @@ def __call__(self, arg):
             pass
         else:
             register()
-            return self(arg)  # recurse
+            return self  # recurse
```

Comment:
that's true, user may not necessarily call it, I added a test to that logic. thanks!